clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -109,6 +112,12 @@ class KubernetesServerGroupCachingAgent extends KubernetesV1CachingAgent impleme
       buildCacheResult([new ReplicaSetOrController(replicationController: replicationController, replicaSet: replicaSet)], [:], [], Long.MAX_VALUE)
     }
 
+    if (result.cacheResults.values().flatten().isEmpty()) {
+      if ( controllerAgent.isControllerAgentSet(namespace, serverGroupName) ) {
```

Comment:
What happens when you have a replica set and a daemon set with the same name?

Reply:
In that case, it might need to depend on the scheduled cache update.

Reply:
won't this update the wrong entry though?